### PR TITLE
Implement repr() for JobConfig

### PR DESCRIPTION
**Pull request by vtomole**
Addressing: https://github.com/quantumlib/Cirq/issues/776. I was thinking about implementing an `__str__`, but  I don't know what that would look like. I think `__repr__` is enough for this class.

### CODE REVIEW

**Comment by vtomole** [comment]
@googlebot  I signed it!

File: cirq/google/engine/engine.py [hunk 1]
```diff
@@ -104,6 +104,19 @@ def copy(self):
             gcs_program=self.gcs_program,
             gcs_results=self.gcs_results)
 
+    def __repr__(self):
+        return ('JobConfig(project_id={!r}, '
+                'program_id={!r}, '
+                'job_id={!r}, '
+                'gcs_prefix={!r}, '
+                'gcs_program={!r}, '
+                'gcs_results={!r}) ').format(self.project_id,
```
Comment (Strilanc) [inline cirq/google/engine/engine.py]: Extra space at end of string

**Comment by vtomole** [comment]
@Strilanc Done.

File: cirq/google/engine/engine.py [hunk 2]
```diff
@@ -104,6 +104,19 @@ def copy(self):
...
             gcs_results=self.gcs_results)
 
+    def __repr__(self):
+        return ('JobConfig(project_id={!r}, '
+                'program_id={!r}, '
+                'job_id={!r}, '
+                'gcs_prefix={!r}, '
+                'gcs_program={!r}, '
+                'gcs_results={!r})').format(self.project_id,
+                                             self.program_id,
```
Comment (Strilanc) [inline cirq/google/engine/engine.py]: Nit: indentation

File: cirq/google/engine/engine_test.py [hunk 3]
```diff
@@ -26,28 +26,41 @@
...
 from cirq.testing.mock import mock
 
+@cirq.testing.only_test_in_python3
+def test_repr():
+    v = cirq.google.JobConfig(project_id='my-project-id',
+                              program_id='my-program-id',
+                              job_id='my-job-id')
+
+    assert repr(v) == ("JobConfig(project_id='my-project-id', "
+                         "program_id='my-program-id', "
```
Comment (Strilanc) [inline cirq/google/engine/engine_test.py]: Nit: indentation

**Comment by vtomole** [comment]
@Strilanc Thanks. Fixed now.

**Pull request merged**